egs2/TEMPLATE/vad1/vad.sh

```diff
@@ -0,0 +1,947 @@
+                _max_length=$(python3 -c "print(int(${max_wav_duration} * ${_fs}))")
+
+                # utt2num_samples is created by format_wav_scp.sh
+                <"${data_feats}/org/${dset}/utt2num_samples" \
+                    awk -v min_length="${_min_length}" -v max_length="${_max_length}" \
+                        '{ if ($2 > min_length && $2 < max_length ) print $0; }' \
+                        >"${data_feats}/${dset}/utt2num_samples"
+                <"${data_feats}/org/${dset}/wav.scp" \
+                    utils/filter_scp.pl "${data_feats}/${dset}/utt2num_samples"  \
+                    >"${data_feats}/${dset}/wav.scp"
+            else
+                # Get frame shift in ms from conf/fbank.conf
+                _frame_shift=
+                if [ -f conf/fbank.conf ] && [ "$(<conf/fbank.conf grep -c frame-shift)" -gt 0 ]; then
+                    # Assume using conf/fbank.conf for feature extraction
+                    _frame_shift="$(<conf/fbank.conf grep frame-shift | sed -e 's/[-a-z =]*\([0-9]*\)/\1/g')"
+                fi
+                if [ -z "${_frame_shift}" ]; then
+                    # If not existing, use the default number in Kaldi (=10ms).
+                    # If you are using different number, you have to change the following value manually.
+                    _frame_shift=10
+                fi
+
+                _min_length=$(python3 -c "print(int(${min_wav_duration} / ${_frame_shift} * 1000))")
+                _max_length=$(python3 -c "print(int(${max_wav_duration} / ${_frame_shift} * 1000))")
+
+                cp "${data_feats}/org/${dset}/feats_dim" "${data_feats}/${dset}/feats_dim"
+                <"${data_feats}/org/${dset}/feats_shape" awk -F, ' { print $1 } ' \
+                    | awk -v min_length="${_min_length}" -v max_length="${_max_length}" \
+                        '{ if ($2 > min_length && $2 < max_length) print $0; }' \
+                        >"${data_feats}/${dset}/feats_shape"
+                <"${data_feats}/org/${dset}/feats.scp" \
+                    utils/filter_scp.pl "${data_feats}/${dset}/feats_shape"  \
+                    >"${data_feats}/${dset}/feats.scp"
+            fi
+
+            # Remove empty text
+            # shellcheck disable=SC2068
+            for ref_txt in ${ref_text_files[@]}; do
+                <"${data_feats}/org/${dset}/${ref_txt}" \
+                    awk ' { if( NF != 1 ) print $0; } ' >"${data_feats}/${dset}/${ref_txt}"
+            done
+
+            # fix_data_dir.sh leaves only utts which exist in all files
+            utils/fix_data_dir.sh \
+                ${ref_text_files_str:+--utt_extra_files "${ref_text_files_str}"} \
+                "${data_feats}/${dset}"
+        done
+
+        if [ -n "${post_process_local_data_opts}" ]; then
+            # Do any additional local data post-processing here
+            local/data.sh ${post_process_local_data_opts} --asr_data_dir "${data_feats}/${train_set}"
```

Comment:
What kind of post-process we would like to do here? I did not find related functions in local/data.sh. I personally. think it would be better to get it generalized and put into pyscripts or scripts for consistency

Reply:
Could you help checking whether we can integrate it in to local/data.sh or it would be better to be designed as a general scripts?